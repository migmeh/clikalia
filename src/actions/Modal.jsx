export const OPEN_MODAL  = 'OPEN_MODAL'
export const CLOSE_MODAL = 'CLOSE_MODAL'

export function showModal(pokemon_name) {
    return {
        type:    OPEN_MODAL,
        payload: pokemon_name
    }
}

export function hideModal() {
    return {
        type: CLOSE_MODAL,
    }
}