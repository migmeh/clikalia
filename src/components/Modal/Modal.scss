@import '../../assets/sass/globals.scss';
@import '../../assets/sass/pokemon_types.scss';

.modal {
    position: fixed;
    top: 0;
    left: 0;
    width:100%;
    height: 100%;
    background: rgba(255, 255, 255, 0.6);
    z-index: 1000;

    .modal-main {
        position: relative;
        background: white;
        width: 250px;
        height: 300px;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
        color: $dark-gray;
        border-radius: 5px;

        .erro {
            padding: 0 40px;
            position: absolute;
            text-align: center;
            top: 50%;
            left: 50%;
            transform: translate(-50%, -50%);
            width: 80%;
        }

        .button {
            display: inline-block;
            border-radius: 5px;
            background-color: #566CD6;
            border: none;
            color: #FFFFFF;
            text-align: center;
            padding: 5px 10px;
            transition: all 0.5s;
            cursor: pointer;
            margin: 5px;
            position: absolute;
            bottom: 0;
            right: 0;
        }

        .pkmn-image {
            position: absolute;
            height: 125px;
            width: auto;
            left: 0;
        }

        .pkmn-info {
            display: grid;
            position: relative;
            float: right;
            text-align: left;
            margin: 20px 2px;
            font-size: 14px;
        }

        .card-type {
            font-size: 12px;
            color: #FFFFFF;
            padding: 10px 10px;
            margin: 0 4px 0 0;
            border-radius: 5px;
        }

        .pkmn-header{
            background-color:  #566CD6;
            height: 25px;
            position: relative;
            border-radius: 5px 5px 0 0;
        }

        .pkmn-title {
            position: fixed;
            color: white;
            text-align: center;
            justify-content: center;
            top: 13px;
            left: 50%;
            transform: translate(-50%, -50%);
            width: 250px;
        }

        .pkmn-types {
            margin-top: 2px;
        }
    }
}

.display-block {
    display: block;
}

.display-none {
    display: none;
}

