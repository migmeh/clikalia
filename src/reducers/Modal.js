const initialState = {
    show_modal:    false,
    modal_pokemon: null,
};
  
export const modal = (state = initialState, { type, payload }) => {
    switch (type) {
      case 'OPEN_MODAL':
          return { ...state, show_modal: true, modal_pokemon: payload }
      case 'CLOSE_MODAL':
          return { ...state, show_modal: false, modal_pokemon: null }
      default:
          return state;
    }
}

export default modal;