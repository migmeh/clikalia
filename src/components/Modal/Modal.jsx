import React from 'react';
import './Modal.scss';

export default class Modal extends React.Component  {
    componentDidMount() {
        this.props.fetchModalPokemon(this.props.modal_pokemon);
    }

    parseTypes = (types) => {
        return (types.map((poketype, index) => {
            return <span key={"card-type "+ poketype.type.name.toLowerCase()} className={"card-type "+ poketype.type.name.toLowerCase()}>
                {poketype.type.name.toUpperCase()}
            </span>
        }))
    }

    parseName = (name) => (
        name.charAt(0).toUpperCase() + name.slice(1)
    );

    parseNumberID = (n) => {
        if (n < 1000 && n >= 100) {
            return n
        } else if (n <= 99 && n > 9) {
            return "0" + n;
        } else {
            return "00" + n;
        }
    }

    render() {
        return (
            <div className={ this.props.show_modal ? "modal display-block" : "modal display-none"}>
                    <section className="modal-main">
                    {(!this.props.error && !this.props.loading_pokemon && this.props.pokemon_info && this.props.pokemon_info.sprites) ?
                        <div>
                            <div className="pkmn-header"></div>
                            <span className="pkmn-title">{this.parseName(this.props.pokemon_info.name)} #{this.props.pokemon_info.id}</span>
                            <img className="pkmn-image" alt="pkmn-modal" src={this.props.pokemon_info.sprites.front_default}/>
                            <div className="pkmn-info">
                                <span>Altura: {this.props.pokemon_info.height*10} cm</span>
                                <span>Peso: {this.props.pokemon_info.weight/10} kg</span>
                                <div className="pkmn-types">{this.parseTypes(this.props.pokemon_info.types)}</div>
                            </div>
                        </div>
                        : <span className="erro">"Error! No encontramos nada"</span>
                    }
                        <button className="button" onClick={() => this.props.hideModal()}>Cerrar</button>
                        </section>
            </div>
        );
    }
}
