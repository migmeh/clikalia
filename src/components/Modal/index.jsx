import { connect } from 'react-redux'
import { 
    hideModal, 
} from '../../actions/Modal'
import {
    fetchModalPokemon,
} from '../../actions/Pokemon'
import Modal from './Modal'

const mapStateToProps = (state) => {
    return {
        show_modal:      state.modal.show_modal,
        modal_pokemon:   state.modal.modal_pokemon,
        pokemon_info:    state.pokemon.modal_pokemon,
        error:           state.pokemon.error,
        loading_pokemon: state.pokemon.fetching,
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        hideModal: () => {
            dispatch(hideModal());
        },
        fetchModalPokemon: (pokemon_name) => {
            dispatch(fetchModalPokemon(pokemon_name));
        },
    }
}


export default connect(mapStateToProps, mapDispatchToProps)(Modal);