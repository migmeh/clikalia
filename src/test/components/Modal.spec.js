import React from 'react';
import { shallow } from 'enzyme';

import Modal from '../../components/Modal/Modal';

describe('Modal', () => {
  it('Should render correctly', () => {
    const tree = shallow(<Modal fetchModalPokemon={jest.fn()}/>);
    expect(tree).toMatchSnapshot();
  });
});