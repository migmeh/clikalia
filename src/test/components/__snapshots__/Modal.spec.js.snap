// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Should render correctly 1`] = `
<div
  className="modal display-none"
>
  <section
    className="modal-main"
  >
    <span
      className="erro"
    >
      "Error! No encontramos nada"
    </span>
    <button
      className="button"
      onClick={[Function]}
    >
      Cerrar
    </button>
  </section>
</div>
`;
